modules/dbnd/src/dbnd/_core/task_run/task_run_sync_local.py

```diff
@@ -164,6 +163,13 @@ def _sync_remote_to_local(self, remote_target, local_target):
         if dbnd_meta_cache.expired or not local_target.exists():
             # If TTL is invalid, or local file doesn't exist -> Sync to local
             local_target.mkdir_parent()
+            _, tmp_file_path = mkstemp()
             remote_target.download(
-                local_target.path, overwrite=local_target.config.overwrite_target,
+                tmp_file_path, overwrite=local_target.config.overwrite_target,
             )
+            try:
+                local_target.move_from(tmp_file_path)
+            except FileAlreadyExists:
+                return
```

Comment:
log.debug message maybe? In case we run into funny stuff

Reply:
also, who is cleaning tmp_file_path?

Reply:
done

Reply:
we probably should remove temp ( with error handling so we don't have double fail) if move fails 

(and that can happen if we have concurrent execution)

Reply:
ah sure

Reply:
fixd